gnome-initial-setup/eos-setup-demo-user

```diff
@@ -0,0 +1,192 @@
+        thing to do, of course, but does not work when pkexec'd by g-i-s:
+        dconf writes happen in a dedicated process, which is service activated
+        over D-Bus; but the pkexec environment has no session bus, and no X11
+        connection with which to auto-launch one. Interestingly there seems to
+        be no way to get an error out of GSettings – you get nice warnings on
+        the console, but nothing here.'''
+        for updates in self.updates.values():
+            updates.apply()
+
+    def write_dconf_compile(self):
+        '''This is a bit roundabout, and depends on the knowledge that the user
+        database lives at $XDG_CONFIG_HOME/dconf/user, but has the advantage
+        of actually working when there's no session bus.'''
+        keyfile_contents = self.format_keyfile()
+
+        with tempfile.TemporaryDirectory(suffix='.d') as d:
+            keyfile = os.path.join(d, '00-live')
+            log.info('writing keyfile to %s', keyfile)
+            with open(keyfile, 'w') as f:
+                f.write(keyfile_contents)
+
+            user_dconf_dir = os.path.join(GLib.get_user_config_dir(), 'dconf')
+            log.info('ensuring %s exists', user_dconf_dir)
+            os.makedirs(user_dconf_dir, exist_ok=True)
+
+            user_dconf = os.path.join(user_dconf_dir, 'user')
+            if os.path.exists(user_dconf):
+                log.error('%s already exists, refusing to overwrite it',
+                          user_dconf)
+                raise SystemExit(1)
+
+            args = ['dconf', 'compile', user_dconf, d]
+            log.info('$ %s', ' '.join(args))
+            subprocess.check_call(args)
+
+    def write_stdout(self):
+        '''Just write the keyfile used by `dconf compile` to stdout.
+        For debugging purposes only.'''
+        print(self.format_keyfile())
+
+    def update_favorite_apps(self):
+        updates = self.updates[SHELL_SCHEMA]
+        favorite_apps = updates.get_settings().get_strv(FAVORITE_APPS_KEY)
+        changed = False
+
+        # Replace Chrome (downloaded on demand) with Chromium (pre-installed).
+        # This effectively undoes a step taken in the image builder,
+        # pre-seeding /var/eos-image-defaults/settings with the opposite
+        # change.
+        if GOOGLE_CHROME in favorite_apps:
+            i = favorite_apps.index(GOOGLE_CHROME)
+            favorite_apps[i] = CHROMIUM_BROWSER
+            changed = True
+
+        if changed:
+            self.update(SHELL_SCHEMA, FAVORITE_APPS_KEY,
+                        GLib.Variant('as', favorite_apps))
+
+    def disallow_app_center_updates(self):
+        self.update(GS_SCHEMA, ALLOW_UPDATES, GLib.Variant('b', False))
+
+
+def main():
+    p = argparse.ArgumentParser(
+        description='Configures system and user settings for demo session')
+    subparsers = p.add_subparsers(title='modes', dest='mode')
+
+    user = subparsers.add_parser(
```

Comment:
Yeah I'd really like to not have another copy of this script. I already have an open task to clean up the duplication between this script and the `copy-worker` script that is used to copy settings from the FBE session to the newly-created user in the normal flow https://phabricator.endlessm.com/T16130 – I was not aware of `copy-worker` when I wrote this script. Its approach is much cleaner because you can just modify the dconf settings of the `gnome-initial-setup` user and they get copied into the new user.